db/migrate/20160408070080_update_follower_relationships_based_on_cloned_from.rb

```diff
@@ -1,101 +1,69 @@
+          ON DUPLICATE KEY UPDATE needs_to_be_deleted = false
+      ")
+      execute("
+        DELETE FROM follower_relationships
+        WHERE needs_to_be_deleted = true
+      ")
     end
+    remove_column :follower_relationships, :needs_to_be_deleted
   end
 
   def down
     ActiveRecord::Base.transaction do
-      # Remove new additions
-      new_relationships =
-        select_all("SELECT fr.id AS fr_id, fr.created_at AS fr_created_at,
-                       p.id AS person_id, p.cloned_from AS person_cloned_from,
-                       f.id AS follower_id, f.cloned_from AS follower_cloned_from
-                       FROM follower_relationships AS fr
-                       LEFT JOIN people AS p ON fr.person_id = p.id
-                       LEFT JOIN people AS f ON fr.follower_id = f.id
-                       WHERE
-                         p.cloned_from IS NOT NULL OR
-                         f.cloned_from IS NOT NULL
-                      ").to_ary
-
-      if new_relationships.present?
-        insert_values = new_relationships.map { |r|
-          p_id = r['person_cloned_from'] || r['person_id']
-          f_id = r['follower_cloned_from'] || r['follower_id']
-          create_insert_values(p_id, f_id, r['fr_created_at'], DateTime.now)
-        }
-        delete_ids = new_relationships.map{ |v| quote(v['fr_id']) }
-
-        insert_values.each_slice(1000) { |batch|
-          execute(create_insert_statement(batch, ignore: true))
-        }
-        delete_ids.each_slice(1000) { |batch|
-          execute(create_delete_statement(batch))
-        }
-      end
+      execute("
+        INSERT IGNORE INTO follower_relationships
+          (person_id, follower_id, created_at, updated_at)
+          SELECT person.cloned_from, follower.cloned_from, fr.created_at, fr.updated_at
+            FROM follower_relationships AS fr
+            LEFT JOIN people AS person ON fr.person_id = person.id
+            LEFT JOIN people AS follower ON fr.follower_id = follower.id
+            WHERE
+              person.cloned_from IS NOT NULL AND
+              follower.cloned_from IS NOT NULL
+          UNION ALL
+          SELECT person.id, follower.cloned_from, fr.created_at, fr.updated_at
+            FROM follower_relationships AS fr
+            LEFT JOIN people AS person ON fr.person_id = person.id
+            LEFT JOIN people AS follower ON fr.follower_id = follower.id
+            WHERE
+              person.cloned_from IS NULL AND
+              follower.cloned_from IS NOT NULL
+          UNION ALL
+            SELECT person.cloned_from, follower.id, fr.created_at, fr.updated_at
+            FROM follower_relationships AS fr
+            LEFT JOIN people AS person ON fr.person_id = person.id
+            LEFT JOIN people AS follower ON fr.follower_id = follower.id
+            WHERE
+              person.cloned_from IS NOT NULL AND
+              follower.cloned_from IS NULL
+      ")
```

Comment:
New relationships are replaced in 3 steps:

1. relationships where both person_id and follower_id are from cloned users
2. relationships where person_id is from an original user and follower_id is from a cloned one
3. relationships where person_id is from a cloned user and follower_id is from an original one